src/analysis/sum_integral_comparisons.lean

```diff
@@ -0,0 +1,169 @@
+that.
+
+At the moment it contains four lemmas in this direction: `antitone_on.integral_le_sum`,
+`antitone_on.sum_le_integral` and versions for monotone functions, which can all be paired
+with a `filter.tendsto` to estimate some errors.
+
+`TODO`: Add more lemmas to the API to directly address limiting issues
+
+## Main Results
+
+* `antitone_on.integral_le_sum`: The integral of an antitone function is at most the sum of its
+  values at integer steps aligning with the left-hand side of the interval
+* `antitone_on.sum_le_integral`: The sum of an antitone function along integer steps aligning with
+  the right-hand side of the interval is at most the integral of the function along that interval
+* `monotone_on.integral_le_sum`: The integral of a monotone function is at most the sum of its
+  values at integer steps aligning with the right-hand side of the interval
+* `monotone_on.sum_le_integral`: The sum of a monotone function along integer steps aligning with
+  the left-hand side of the interval is at most the integral of the function along that interval
+
+## Tags
+
+analysis, comparison, asymptotics
+-/
+
+open set measure_theory.measure_space
+open_locale big_operators
+
+lemma antitone_on.integral_le_sum {x₀ : ℝ} {a : ℕ} {f : ℝ → ℝ}
+  (hf : antitone_on f (Icc x₀ (x₀ + a))) :
+  ∫ x in x₀..(x₀ + a), f x ≤ ∑ i in finset.range a, f (x₀ + i) :=
+begin
+  have hint : ∀ (k : ℕ), k < a → interval_integrable f volume (x₀+k) (x₀ + (k + 1 : ℕ)),
+  { assume k hk,
+    refine (hf.mono _).interval_integrable,
+    rw interval_of_le,
+    { apply Icc_subset_Icc,
+      { simp only [le_add_iff_nonneg_right, nat.cast_nonneg] },
+      { simp only [add_le_add_iff_left, nat.cast_le, nat.succ_le_of_lt hk] } },
+    { simp only [add_le_add_iff_left, nat.cast_le, nat.le_succ] } },
+  calc
+  ∫ x in x₀..(x₀ + a), f x = ∑ i in finset.range a, ∫ x in (x₀+i)..(x₀+(i+1 : ℕ)), f x :
+    begin
+      convert (interval_integral.sum_integral_adjacent_intervals hint).symm,
+      simp only [nat.cast_zero, add_zero],
+    end
+  ... ≤ ∑ i in finset.range a, ∫ x in (x₀+i)..(x₀+(i+1)), f (x₀ + i) :
+    begin
+      apply finset.sum_le_sum (λ i hi, _),
+      have ia : i < a := finset.mem_range.1 hi,
+      refine interval_integral.integral_mono_on (by simp) (hint _ ia) (by simp) (λ x hx, _),
+      apply hf _ _ hx.1,
+      { simp only [ia.le, mem_Icc, le_add_iff_nonneg_right, nat.cast_nonneg, add_le_add_iff_left,
+          nat.cast_le, and_self] },
+      { refine mem_Icc.2 ⟨le_trans (by simp) hx.1, le_trans hx.2 _⟩,
+        simp only [add_le_add_iff_left, nat.cast_le, nat.succ_le_of_lt ia] },
+    end
+  ... = ∑ i in finset.range a, f (x₀ + i) : by simp
+end
+
+lemma antitone_on.integral_le_sum_Ico {a b : ℕ} {f : ℝ → ℝ} (hab : a ≤ b)
+  (hf : antitone_on f (set.Icc a b)) : ∫ x in a..b, f x ≤ ∑ x in finset.Ico a b, f x :=
+begin
+  have bb : b = (b - a) + a, { zify, ring, },
```

Comment:
Does this work?
```suggestion
  have bb : b = (b - a) + a := (nat.sub_add_cancel hab).symm,
```
Better: don't define `bb` at all, but replace it in the `rw`.

Reply:
OK, this required some extra `conv`'s but maybe this is nicer?

Reply:
I don't see why it required extra `conv`. You had `have bb : ...` followed by, somewhere in the next line, `rw bb`. I was suggesting to remove the `have bb` and replace that `rw bb` by `rw ← nat.sub_add_cancel hab`, which should do exactly the same thing.

But the proof you have now is fine. Let's not spend time discussing/changing such a detail.

Reply:
For reasons I don't quite understand, in one version the rewriter only changed one instance; in the second version, it changed all instances. 🤷